app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -17,85 +16,43 @@ import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.ui.unit.sp
-
-private val homeScreenMapper: HomeScreenMapper = HomeScreenMapperImpl()
-
-val popularCategoryLabels = listOf(
-    MovieCategory.POPULAR_STREAMING,
-    MovieCategory.POPULAR_ONTV,
-    MovieCategory.POPULAR_FORRENT,
-    MovieCategory.POPULAR_INTHEATHERS
-)
-
-val nowPlayingCategoryLabels = listOf(
-    MovieCategory.NOWPLAYING_MOVIES,
-    MovieCategory.NOWPLAYING_TV
-)
-
-val upcomingCategoryLabels = listOf(
-    MovieCategory.UPCOMING_TODAY,
-    MovieCategory.UPCOMING_THISWEEK
-)
-
-val popularCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
-    popularCategoryLabels,
-    MovieCategory.POPULAR_STREAMING,
-    MoviesMock.getMoviesList()
-)
-val nowPlayingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
-    nowPlayingCategoryLabels,
-    MovieCategory.NOWPLAYING_MOVIES,
-    MoviesMock.getMoviesList()
-)
-val upcomingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
-    upcomingCategoryLabels,
-    MovieCategory.UPCOMING_TODAY,
-    MoviesMock.getMoviesList()
-)
+import kotlinx.coroutines.Dispatchers
 
 @Composable
 fun HomeScreenRoute(
+    viewModel: HomeViewModel,
     onNavigateToMovieDetails: (String) -> Unit,
-    onFavoriteButtonClicked: () -> Unit
 ) {
-    var popularViewState by remember { mutableStateOf(popularCategoryViewState) }
-    var nowPlayingViewState by remember { mutableStateOf(nowPlayingCategoryViewState) }
-    var upcomingViewState by remember { mutableStateOf(upcomingCategoryViewState) }
+    val popularViewState: HomeMovieCategoryViewState by viewModel.popularViewState.collectAsState()
+    val nowPlayingViewState: HomeMovieCategoryViewState by viewModel.nowPlayingViewState.collectAsState()
+    val upcomingViewState: HomeMovieCategoryViewState by viewModel.upcomingViewState.collectAsState()
 
     HomeScreen(
         popularViewState = popularViewState,
         nowPlayingViewState = nowPlayingViewState,
         upcomingViewState = upcomingViewState,
         onPopularLabelClick = {
-            popularViewState = homeScreenMapper.toHomeMovieCategoryViewState(
-                popularCategoryLabels,
-                MovieCategory.getByOrdinal(it)!!,
-                MoviesMock.getMoviesList()
-            )
+            viewModel.onPopularLabelClick(it)
         },
         onNowPlayingLabelClick = {
-            nowPlayingViewState = homeScreenMapper.toHomeMovieCategoryViewState(
-                nowPlayingCategoryLabels,
-                MovieCategory.getByOrdinal(it)!!,
-                MoviesMock.getMoviesList()
-            )
+            viewModel.onNowPlayingLabelClick(it)
         },
         onUpcomingLabelClick = {
-            upcomingViewState = homeScreenMapper.toHomeMovieCategoryViewState(
-                upcomingCategoryLabels,
-                MovieCategory.getByOrdinal(it)!!,
-                MoviesMock.getMoviesList()
-            )
+            viewModel.onUpcomingLabelClick(it)
```

Comment:
There should be just one `onMovieCategoryClick` parameter which will take category id. No need for multiple functions